lib/ansible/plugins/vars/recursive_group_vars.py

```diff
@@ -0,0 +1,95 @@
+        - If several matching directories are found on the same level they are traversed consecutively in a depth-first fashion.
+        - The deepest found matching file is applied last.
+        - Only files that match a group name are applied. Any other files are ignored.
+        - The following points are identical to host_group_vars plugin:
+            - Files are restricted by extension to one of .yaml, .json, .yml or no extension.
+            - Hidden (starting with '.') and backup (ending with '~') files and directories are ignored.
+            - Only applies to inventory sources that are existing paths.
+    options:
+      _valid_extensions:
+        default: [".yml", ".yaml", ".json"]
+        description:
+          - "Check all of these extensions when looking for 'variable' files which should be YAML or JSON or vaulted versions of these."
+          - 'This affects vars_files, include_vars, inventory and vars plugins among others.'
+        env:
+          - name: ANSIBLE_YAML_FILENAME_EXT
+        ini:
+          - section: yaml_valid_extensions
+            key: defaults
+        type: list
+'''
+
+import os
+from ansible.module_utils._text import to_bytes, to_native, to_text
+from ansible.errors import AnsibleParserError
+from ansible.plugins.vars import BaseVarsPlugin
+from ansible.inventory.group import Group
+from ansible.utils.vars import combine_vars
+
+FOUND = {}
+
+class VarsModule(BaseVarsPlugin):
+
+    def find_vars_recurse(self, loader, groups, path):
+        data = {}
+
+        self._display.debug("recursive_group_vars - Traversing dir : " + path + " with groups : " + to_text(groups))
+
+        # Apply matching files in path
+        for group in groups:
+            try:
+                found_files = loader.find_vars_files(path, group.name, allow_dir=False)
+                for found in found_files:
+                    self._display.debug("recursive_group_vars - Matched file : " + to_text(found))
+                    new_data = loader.load_from_file(found, cache=True, unsafe=True)
+                    if new_data:
+                        data = combine_vars(data, new_data)
+            except Exception as e:
+                raise AnsibleParserError(to_native(e))
+
+        # Recurse into directories
+        for group in groups:
+            try:
+                b_opath = os.path.realpath(to_bytes(os.path.join(path, group.name)))
+                if os.path.exists(b_opath):
+                    if os.path.isdir(b_opath):
+                        data = combine_vars(data,
+                                self.find_vars_recurse(loader, groups, to_text(b_opath)))
+            except Exception as e:
+                raise AnsibleParserError(to_native(e))
+
+        return data
+
+    def get_vars(self, loader, path, entities, cache=True):
+
+        if not isinstance(entities, list):
+            entities = [entities]
+
+        groups = list(filter(lambda ent : isinstance(ent, Group), entities))
```

Comment:
Please convert this to just a list comprehension:

```suggestion
        groups = [e.name for e in entities if isinstance(e, Group)]
```

Additionally, just convert to a list of names here, since you only ever need that, and not the object later.